core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -277,27 +280,93 @@ func (r *EvmRegistry) refreshActiveUpkeeps() error {
 //
 // TODO: check for updated config for log trigger upkeeps and update it, currently we ignore them.
 func (r *EvmRegistry) refreshLogTriggerUpkeeps(ids []*big.Int) error {
-	logTriggerIDs := make([]*big.Int, 0)
+	var err error
+	for i := 0; i < len(ids); i += logTriggerRefreshBatchSize {
+		end := i + logTriggerRefreshBatchSize
+		if end > len(ids) {
+			end = len(ids)
+		}
+		batch := ids[i:end]
+
+		if batchErr := r.refreshLogTriggerUpkeepsBatch(batch); batchErr != nil {
+			multierr.AppendInto(&err, batchErr)
+		}
+
+		time.Sleep(500 * time.Millisecond)
+	}
+
+	return err
+}
+
+func (r *EvmRegistry) refreshLogTriggerUpkeepsBatch(ids []*big.Int) error {
+	var logTriggerIDs []*big.Int
+	var logTriggerHashes []common.Hash
 	for _, id := range ids {
 		uid := &ocr2keepers.UpkeepIdentifier{}
 		if ok := uid.FromBigInt(id); !ok {
 			r.lggr.Warnf("failed to parse upkeep id %s", id.String())
+			continue
 		}
 		switch core.GetUpkeepType(*uid) {
 		case ocr2keepers.LogTrigger:
 			logTriggerIDs = append(logTriggerIDs, id)
-		default:
+			logTriggerHashes = append(logTriggerHashes, common.BigToHash(id))
 		}
 	}
 	newUpkeeps, err := r.logEventProvider.RefreshActiveUpkeeps(logTriggerIDs...)
```

Comment:
just realized, current implementation of `r.logEventProvider.RefreshActiveUpkeeps` expects a list of all active upkeep IDs as it removes the upkeeps it doesn't find in the list. I think we'll need to do this step as a higher level call and then start the batch updates for fetching logs

Reply:
yep, basically start the batching logic after this step.